tools/niminst/niminst.nim

```diff
@@ -769,6 +770,51 @@ proc checkedShellExec(cmd: string) =
   if exitCode != 0:
     raise newExternalProgramError(exitCode)
 
+proc computeChecksum(file: string): string =
+  ## Calculate the SHA256 hash for `file`.
+  let p = startProcess(
+    "openssl",
+    args = ["dgst", "-sha256", quoteShell(file)],
+    options = {poUsePath, poStdErrToStdOut, poEchoCmd}
+  )
+
+  defer: close p
+
+  let exitCode = p.waitForExit()
+
+  if exitCode != 0:
+    stderr.write p.outputStream.readAll()
+    raise newExternalProgramError(exitCode)
+  else:
+    # OpenSSL produce an output in this format:
+    #
+    #   algo(filename)= checksum
+    #
+    # First separate the checksum out
+    result = p.outputStream.readAll().rsplit('=', maxsplit = 2)[1]
+    # Then trim trailing space on both sides
+    result = result.strip()
+
+type
+  ArchiveKind {.pure.} = enum
+    Source = "source"
+    Binary = "binary"
+
+  ArchiveManifest = object
+    ## The generated archive manifest. This object is used as the schema for
+    ## the generated JSON, so changes should be reflected to its users
+    ## (currently release_maker).
+    name: string ## The archive file name
+    sha256: string ## The archive checksum
+    version: string ## The project version
+    case kind: ArchiveKind ## The archive type, in the generated JSON this field
+                           ## will be a string.
+    of Binary:
+      os: string ## The binary archive target OS
+      cpu: string ## The binary archive target CPU
+    of Source:
+      discard
+
 proc archiveDist(c: var ConfigData) =
```

Comment:
```suggestion
proc createArchiveDist(c: var ConfigData) =
```

If we only day it a few times then it's not worth saving the characters and it completely obviates the comment.